build.gradle

```diff
@@ -170,9 +171,19 @@ dependencies {
 def opensearch_tmp_dir = rootProject.file('build/private/opensearch_tmp').absoluteFile
 opensearch_tmp_dir.mkdirs()
 
+task windowsPatches(type:Exec) {
+    commandLine 'cmd', '/c', "Powershell -File $rootDir\\scripts\\windowsScript.ps1"
+}
+
 task cmakeJniLib(type:Exec) {
     workingDir 'jni'
-    commandLine 'cmake', '.', "-DKNN_PLUGIN_VERSION=${opensearch_version}"
+    if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+        dependsOn windowsPatches
+        commandLine 'cmake', '.', "-G", "Unix Makefiles", "-DKNN_PLUGIN_VERSION=${opensearch_version}", "-DCXX_COMPILER_VERSION=12.2.0", "-DBLAS_LIBRARIES=$rootDir\\src\\main\\resources\\windowsDependencies\\libopenblas.dll", "-DLAPACK_LIBRARIES=$rootDir\\src\\main\\resources\\windowsDependencies\\libopenblas.dll"
```

Comment:
Is there anyway we can avoid hardcoding DCXX_COMPILER_VERSION?

Reply:
updated